Billing: add site-level billing section nav

michaeldcain opened the pull request:
This adds a sectionsNav to the new site-level billing screens. It includes #46175, which adds a second "section" (Billing History) to site-level billing. Those commits will be rebased out after #46175 merges.

We should hold on merging this until #46224 is complete.

<img width="1486" alt="Screen Shot 2020-10-07 at 11 35 06 AM" src="https://user-images.githubusercontent.com/942359/95353354-2e008c00-0891-11eb-9280-b9024e5faa4b.png">

Fixes: #46210 

**To test:**
- Navigate to the new site-level billing section: _Plan > Billing_ in the sidebar
- Verify that the sectionNav is displayed
- Verify that the "Purchases" item is selected by default
- Verify that clicking "Billing History" takes you to the site-level billing history screen
- Verify that clicking "Purchases" takes you back to the subscriptions screen
- Verify that you don't see a link to "View all billing history" at the bottom of the subscriptions screen

a8ci18n commented on client/my-sites/purchases/main.tsx:
```diff
@@ -200,3 +209,30 @@ export function PurchaseCancelDomain( {
+
+			<BillingHistoryList siteId={ selectedSiteId } />
+			<CompactCard href="/me/purchases/billing">
+				{ translate( 'View all billing history and receipts' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'View billing history and receipts' )` *ES Score: 10*

fditrapani commented:
Very nice! Just to be clear we'll be holding off on merging this until the receipts on the billing history are ready right?

michaeldcain commented:
Rebased out #46175.

> Just to be clear we'll be holding off on merging this until the receipts on the billing history are ready right?

Yes, I added a note to the description.

fditrapani commented:
Awesome. Thanks for adding that!

krymson24 commented:
@michaeldcain looks great!

Step 2 and 3 Verification:
![2020-10-07 22 11 59](https://user-images.githubusercontent.com/66652282/95407119-2d481400-08ea-11eb-9703-9f02730df450.gif)

Step 4 Verification:
<img width="1105" alt="Screen Shot 2020-10-07 at 10 10 11 PM" src="https://user-images.githubusercontent.com/66652282/95407187-536db400-08ea-11eb-9f2c-dfd9656dbe50.png">

Step 5 and 6 Verification:
<img width="1100" alt="Screen Shot 2020-10-07 at 10 10 46 PM" src="https://user-images.githubusercontent.com/66652282/95407217-61233980-08ea-11eb-8b87-499d3d1af4e9.png">

krymson24 commented on client/my-sites/purchases/main.tsx:
```diff
@@ -223,6 +227,8 @@ export function BillingHistory() {
 				align="left"
 			/>
+			<PurchasesNavigation sectionTitle={ 'Billing History' } siteSlug={ siteSlug } />
+
```
Unnecessary space?

michaeldcain replied:
Thanks for the review! For this, I was following [the pattern elsewhere in this file](https://github.com/Automattic/wp-calypso/pull/46246/files/dec23eb956fd2c4f682f1a98e91d4700a60c8aad#diff-9707f7ea120cf90a339783e4157e9394L48), where there was an empty space between more of the "header" parts and the "content" of the component. Those might also be typos, but I also kind of liked the readability.

krymson24 replied:
@michaeldcain non-blocker! (hence the approval) -- do what you feel is best!

krymson24 commented on client/my-sites/purchases/navigation.tsx:
```diff
@@ -0,0 +1,41 @@
+export default function PurchasesNavigation( {
+	sectionTitle,
+	siteSlug,
+}: {
```
Hmm, didn't know our linting would let this through 🤔

michaeldcain replied:
@krymson24: could you expound? I'm still pretty new to TypeScript, but this seems to be pretty standard syntax.

krymson24 replied:
This just looked weird to me -- but I think it's correct. Carry on!

michaeldcain commented:
Thanks for the review @krymson24! Just waiting on #46224 to merge, since this will effectively launch the feature.

sirbrillig commented:
Sorry that I caused you a conflict. I moved the billing-history functions from `main.tsx` to `billing-history/index.tsx`.

sirbrillig reviewed (approved):
Code looks good to me, and the testing instructions work as described. Nice work!

fditrapani reviewed (approved):
Gave this another spin and it's looking ready to roll  🎲 🎲 🎲
